torch/_inductor/fx_passes/mkldnn_fusion.py

```diff
@@ -413,47 +450,101 @@ def __init__(self, op_name: str, scalars_attr=None, algorithm_attr=None):
-            UnaryAttr("tanh"): [
-                _combined_fusion(u, aten.tanh) for u in _computation_user_1
-            ],
-        }
-        for unary_attr, patterns in replacement_unary_fusion_patterns.items():
-            _register_unary_fusion_lowering(patterns[0], unary_attr, computation_ops[0])
-            _register_unary_fusion_lowering(patterns[1], unary_attr, computation_ops[1])
-            _register_unary_fusion_lowering(patterns[2], unary_attr, computation_ops[2])
+        call_user1 = [
+            _conv_call(users=1),
+            _linear_call(users=1),
+            _conv_transpose_call(users=1),
+        ]
+        call_user3 = [
+            _conv_call(users=3),
+            _linear_call(users=3),
+            _conv_transpose_call(users=3),
+        ]
 
+        def _fp32_fusion_patterns():
+            call_user2 = [
+                _conv_call(users=2),
+                _linear_call(users=2),
+                _conv_transpose_call(users=2),
+            ]
+            call_user4 = [
+                _conv_call(users=4),
+                _linear_call(users=4),
+                _conv_transpose_call(users=4),
+            ]
+            replacement_unary_fusion_patterns = {
+                UnaryAttr("gelu", algorithm_attr="tanh"): [
+                    _gelu_fusion_2(u) for u in call_user4
+                ],
+                UnaryAttr("gelu", algorithm_attr="none"): [
+                    _gelu_fusion_1(u) for u in call_user2
+                ],
+                UnaryAttr("hardswish"): [_hardswish_fusion(u) for u in call_user2],
+                UnaryAttr("hardsigmoid"): [_hardsigmoid_fusion(u) for u in call_user1],
+                UnaryAttr("swish"): [_silu_fusion(u) for u in call_user2],
+                UnaryAttr("relu"): [_combined_fusion(u, aten.relu) for u in call_user1],
+                UnaryAttr("sigmoid"): [
+                    _combined_fusion(u, aten.sigmoid) for u in call_user1
+                ],
+                UnaryAttr("tanh"): [_combined_fusion(u, aten.tanh) for u in call_user1],
+            }
+            return replacement_unary_fusion_patterns
+
+        def _bf16_fusion_patterns():
+            replacement_unary_fusion_patterns = {
+                UnaryAttr("gelu", algorithm_attr="tanh"): [
+                    _bf16_fusion(u, _gelu_fusion_2, users=4) for u in call_user1
+                ],
+                UnaryAttr("gelu", algorithm_attr="none"): [
+                    _bf16_fusion(u, _gelu_fusion_1, users=2) for u in call_user1
+                ],
+                UnaryAttr("hardswish"): [
+                    _bf16_fusion(u, _hardswish_fusion, users=2) for u in call_user1
+                ],
+                UnaryAttr("hardsigmoid"): [
+                    _bf16_fusion(u, _hardsigmoid_fusion, users=1) for u in call_user1
+                ],
+                UnaryAttr("swish"): [
+                    _bf16_fusion(u, _silu_fusion, users=2) for u in call_user1
+                ],
```

Comment:
tanh is not here? Also can we avoid duplicate code by directly constructing bf16 patterns with fp32 list?

Reply:
tanh is not decomposed.